modules/consensus/synchronize.go

```diff
@@ -113,27 +126,39 @@ func (s *ConsensusSet) sendBlocks(conn modules.PeerConn) error {
 	// Find the most recent block from knownBlocks in the current path.
 	found := false
 	var start types.BlockHeight
-	lockID := s.mu.RLock()
-	if !s.db.open {
-		s.mu.RUnlock(lockID)
-		return errors.New("database not open")
-	}
-	for _, id := range knownBlocks {
-		if s.db.inBlockMap(id) {
-			pb := s.db.getBlockMap(id)
-			if pb.Height <= s.height() && id == s.db.getPath(pb.Height) {
-				found = true
-				start = pb.Height + 1 // start at child
+	var csHeight types.BlockHeight
+	lockID := cs.mu.RLock()
+	err = cs.db.View(func(tx *bolt.Tx) error {
+		csHeight = blockHeight(tx)
+		for _, id := range knownBlocks {
+			pb, err := getBlockMap(tx, id)
+			if err != nil {
+				continue
+			}
+			pathID, err := getPath(tx, pb.Height)
+			if err != nil {
+				continue
+			}
+			if pathID != pb.Block.ID() {
+				continue
+			}
+			if pb.Height == blockHeight(tx) {
```

Comment:
haven't you already calculated the blockHeight as `csHeight`?

also, would probably help to have a function that fetched a block from current path by ID